skyportal/handlers/api/observation_plan.py

```diff
@@ -420,8 +433,211 @@ def get(self, observation_plan_request_id):
+            suggested filename based on `source_name` and `output_format`
+        data : str
+            binary encoded data for the movie (to be streamed)
+        reason : str
+            If not successful, a reason is returned.
+    """
+
+    surveyColors = {
+        "ztfg": "#28A745",
+        "ztfr": "#DC3545",
+        "ztfi": "#F3DC11",
+        "AllWISE": "#2F5492",
+        "Gaia_EDR3": "#FF7F0E",
+        "PS1_DR1": "#3BBED5",
+        "GALEX": "#6607C2",
+        "TNS": "#ED6CF6",
+    }
+
+    filters = list(set(obs.filt for obs in observations))
+    for filt in filters:
+        if filt in surveyColors:
+            continue
+        surveyColors[filt] = "#" + ''.join(
+            [random.choice('0123456789ABCDEF') for i in range(6)]
+        )
+
+    matplotlib.use("Agg")
+    fig = plt.figure(figsize=figsize, constrained_layout=False)
+    ax = plt.axes(projection='astro mollweide')
+    ax.imshow_hpx(localization.flat_2d, cmap='cylon')
+
+    old_artists = []
+
+    def plot_schedule(k):
+        for artist in old_artists:
+            artist.remove()
+        del old_artists[:]
+
+        for i, obs in enumerate(observations):
+
+            if decay != 0:
+                alpha = np.exp((i - k) / decay)
+            else:
+                alpha = alpha_default
+            if alpha > 1:
+                alpha = 1
+
+            if alpha > alpha_cutoff:
+                coords = obs.field.contour_summary["features"][0]["geometry"][
+                    "coordinates"
+                ]
+                ras = np.array(coords)[:, 0]
+                # cannot handle 0-crossing well
+                if len(np.where(ras > 180)[0]) > 0 and len(np.where(ras < 180)) > 0:
+                    continue
+                poly = plt.Polygon(
+                    coords,
+                    alpha=alpha,
+                    facecolor=surveyColors[obs.filt],
+                    edgecolor='black',
+                    transform=ax.get_transform('world'),
+                )
+                ax.add_patch(poly)
+                old_artists.append(poly)
+
+        patches = []
+        for filt in filters:
+            patches.append(mpatches.Patch(color=surveyColors[filt], label=filt))
+        plt.legend(handles=patches)
+
+    if output_format == "gif":
+        writer = animation.PillowWriter()
+    elif output_format == "mp4":
+        writer = animation.FFMpegWriter()
+    else:
+        raise ValueError('output_format must be gif or mp4')
+
+    with tempfile.NamedTemporaryFile(mode='w', suffix='.' + output_format) as f:
+        anim = animation.FuncAnimation(fig, plot_schedule, frames=len(observations))
+        anim.save(f.name, writer=writer)
+        f.flush()
+
+        with open(f.name, mode='rb') as g:
+            anim_content = g.read()
+
+    return {
+        "success": True,
+        "name": f"{localization.localization_name}.{output_format}",
+        "data": anim_content,
+        "reason": "",
+    }
+
```

Comment:
Honestly I didn't understand much of this function but I trust you've kicked the tires on it, so it is probably ok.

Reply:
yeah it works ok. Will probably ask a student to beautify it more down the road once it gets some more use.